Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp

```diff
@@ -204,17 +204,16 @@ class GPUProxyForCapture final : public UserMediaCaptureManagerProxy::Connection
             if (!m_process.get()->allowsVideoCapture())
                 return false;
 #if PLATFORM(IOS_FAMILY)
-            MediaSessionManageriOS::providePresentingApplicationPID();
+            MediaSessionHelper::sharedHelper().providePresentingApplicationPID(m_process.get()->presentingApplicationPID(pageIdentifier));
```

Comment:
Maybe assert that we have a pid in the map.
What would happen if there was no pid in the map?
Should we protect from setting the presentation PID if it is zero?

Reply:
It should not be possible in practice to get a PID of 0 here, because `presentingApplicationAuditToken()` (which is called by `presentingApplicationPID()`) falls back to the parent process audit token if it can't find a per-Page audit token. The only case where a PID of 0 would be returned is if the parent process connection is null, which I think would mean the GPU process is just about to exit.

I can assert the PID is non-null here.